scripts/installer/curl_install_pypi/install.py

```diff
@@ -0,0 +1,420 @@
+        print_status()
+    else:
+        print_status("You can run the promptflow with '{}'.".format(exec_filepath))
+
+
+def verify_python_version():
+    print_status('Verifying Python version.')
+    v = sys.version_info
+    if v < (3, 8):
+        raise CLIInstallError('The promptflow does not support Python versions less than 3.8.')
+    if 'conda' in sys.version:
+        raise CLIInstallError("This script does not support the Python Anaconda environment. "
+                              "Create an Anaconda virtual environment and install with 'pip'")
+    print_status('Python version {}.{}.{} okay.'.format(v.major, v.minor, v.micro))
+
+
+def _native_dependencies_for_dist(verify_cmd_args, install_cmd_args, dep_list):
+    try:
+        print_status("Executing: '{} {}'".format(' '.join(verify_cmd_args), ' '.join(dep_list)))
+        subprocess.check_output(verify_cmd_args + dep_list, stderr=subprocess.STDOUT)
+        print_status('Native dependencies okay.')
+    except subprocess.CalledProcessError:
+        err_msg = 'One or more of the following native dependencies are not currently installed and may be required.\n'
+        err_msg += '"{}"'.format(' '.join(install_cmd_args + dep_list))
+        print_status(err_msg)
+        ans_yes = prompt_y_n('Missing native dependencies. Attempt to continue anyway?', 'n')
+        if not ans_yes:
+            raise CLIInstallError('Please install the native dependencies and try again.')
+
+
+def _get_linux_distro():
+    if platform.system() != 'Linux':
+        return None, None
+
+    try:
+        with open('/etc/os-release') as lines:
+            tokens = [line.strip() for line in lines]
+    except Exception:
+        return None, None
+
+    release_info = {}
+    for token in tokens:
+        if '=' in token:
+            k, v = token.split('=', 1)
+            release_info[k.lower()] = v.strip('"')
+
+    return release_info.get('name', None), release_info.get('version_id', None)
+
+
+def verify_native_dependencies():
+    distname, version = _get_linux_distro()
+
+    if not distname:
+        # There's no distribution name so can't determine native dependencies required / or they may not be needed
+        # like on OS X
+        return
+
+    print_status('Verifying native dependencies.')
+    # is_python3 = sys.version_info[0] == 3
+    distname = distname.lower().strip()
+
+    verify_cmd_args, install_cmd_args, dep_list = None, None, None
+    if any(x in distname for x in ['ubuntu', 'debian']):
+        verify_cmd_args = ['dpkg', '-s']
+        install_cmd_args = ['apt-get', 'update', '&&', 'apt-get', 'install', '-y']
+        python_dep = 'python3-dev'
+        if distname == 'ubuntu' and version in ['12.04', '14.04'] or distname == 'debian' and version.startswith('7'):
+            dep_list = ['libssl-dev', 'libffi-dev', python_dep]
+        elif (distname == 'ubuntu' and version in ['15.10', '16.04', '18.04'] or distname == 'debian' and
```

Comment:
currently ubuntu is 22.10

Reply:
Not sure whether we only check LTS only. 

Do we need check every version of ubuntu and linux distribution? In fact, I'm not sure whether these validation conditions are complete. 
At least in ubuntu20.04 version. It seems that there is no python3-dev (there is python3), and there is no libffi-dev, but it can be installed successfully and run pf flow test.

CC: @wangchao1230

Reply:
we might remove this check for now